tensordict/nn/common.py

```diff
@@ -58,14 +59,78 @@ def _check_all_nested(list_of_keys):
 
 
 def dispatch_kwargs(func):
+    """Allows for a function expecting a TensorDict to be called using kwargs.
+
+    This method must be used within modules that have an :obj:`in_keys` and
+    :obj:`out_keys` attributes indicating what keys to be read and written
+    from the tensordict. The wrapped function should also have a :obj:`tensordict`
+    leading argument.
+
+    The resulting function will return a single tensor (if there is a single
+    element in out_keys), otherwise it will return a tuple sorted as the :obj:`out_keys`
+    of the module.
+
+    Examples:
+        >>> class MyModule(nn.Module):
+        ...     in_keys = ["a"]
+        ...     out_keys = ["b"]
+        ...
+        ...     @dispatch_kwargs
+        ...     def forward(self, tensordict):
+        ...         tensordict['b'] = tensordict['a'] + 1
+        ...         return tensordict
+        ...
+        >>> module = MyModule()
+        >>> b = module(a=torch.zeros(1, 2))
+        >>> assert (b == 1).all()
+
+    For nested keys, it is assumed that an underscore (`_`) is used to join the
+    sub-keys.
+
+    Examples:
+        >>> class MyModuleNest(nn.Module):
+        ...     in_keys = [("a", "c")]
+        ...     out_keys = ["b"]
+        ...
+        ...     @dispatch_kwargs
+        ...     def forward(self, tensordict):
+        ...         tensordict['b'] = tensordict['a', 'c'] + 1
+        ...         return tensordict
+        ...
+        >>> module = MyModuleNest()
+        >>> b, = module(a_c=torch.zeros(1, 2))
+        >>> assert (b == 1).all()
+
+
+    """
+
+    # sanity check
```

Comment:
@tcbegley what do you think of this?
One issue with tensordict is that we name the package like we name the objects, it's not ideal.
Besides enforcing that the first arg always has the same name can be confusing.
We could check that the first input is typed as being a TensorDictBase but even that won't solve the problem (we're not very restrictive on types, it would be surprising to do that here no?)